packages/amplify_core/lib/src/types/exception/storage/operation_canceled_exception.dart

```diff
@@ -0,0 +1,19 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@template amplify_core.storage.operation_canceled_exception}
+/// Exception thrown when a storage operation is canceled.
+/// {@endtemplate}
+class StorageOperationCanceled extends StorageException {
```

Comment:
I think we should come together and formalize our policy for category exception naming.  

In Analytics I didn't prefix exceptions with 'Analytics'. 

For Auth, it seems we do and we don't sometimes.  For example: 

ie. InvalidStateException, UserCancelledException, etc.
but not AuthValidationException, etc. 

Perhaps I'm missing why we do/don't prefix with Auth? 

Regardless isn't it more clear to always prefix with Auth/Analytics/Storage? 

Thoughts? 
 @dnys1 @Jordan-Nelson

Reply:
We added the prefix mainly to align with amplify-swift, although Swift can do an enum over an error type, e.g.

```
StorageError.unknown
StorageError.keyNotFound
```

We don't have this support on the same pattern (yet) for caching the specific exception types, so we have to keep them separate, but in the meantime, we still want to maintain the grouping.

Reply:
If I understand correctly, this exception is thrown when a `StorageOperation` is cancelled. If it was thrown when an `UploadOperation` was canceled it might be called an `UploadOperationCanceledException`. So I am not sure that the `Storage` prefix is really coming from the category in this case. I think it is describing the the operation.

It would be nice to follow a common pattern. I don't think we need to hold this PR up for it, but we can continue discussion.